### ISSUE TITLE
transform-classes loose cannot extend null

### ISSUE THREAD

**Issue by jridgewell** [babel/babel#7539]
Choose one: is this a bug report or feature request?

Bug.

When using transforming classes loosely, you can't subclass `null`.

### Input Code

```js
class Test extends null {
}
```

[repl](http://babeljs.io/repl/build/master#?babili=false&browsers=ie%2010&build=&builtIns=usage&code_lz=MYGwhgzhAEAqCmEAu14A8nwHYBMZYFcQRoBvAKAF8g&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=true&fileSize=false&lineWrap=true&presets=es2015-loose&prettier=false&targets=&version=7.0.0-beta.40&envVersion=)

### Babel/Babylon Configuration (.babelrc, package.json, cli command)

```json
{
  "plugins": ["transform-classes", { "loose": true }]
}
```

Or,

```json
{
  "presets": ["es2015", { "loose": true }]
}
```

### Expected Behavior

Should create a new class that has `null` as "superclass" (ie, it doesn't extend from the default `Object.prototype`).

### Current Behavior

Throws `Cannot read property 'prototype' of null`

### Possible Solution

The `inheritsLoose` [helper](https://github.com/babel/babel/blob/8e030e28b350406d7ad6ce5a44f47d0d60e33509/packages/babel-helpers/src/helpers.js#L438-L444) needs to guard against a `null` superclass.

**Comment by Kovensky** [comment]
Don't worry, spec can't extend null either. Or at least it's not supposed to work.

It's a spec bug that has been there since ES2015.

The only classes that extend null that will not throw at runtime are classes that return an unrelated object from their constructor.

**Comment by loganfsmyth** [comment]
Yeah as far as I know your expected behavior is not actually the expected behavior defined by the spec.

**Comment by Kovensky** [comment]
<img width="555" alt="image" src="https://user-images.githubusercontent.com/73085/37261186-e15d2806-25dd-11e8-9af4-858b6cb72879.png">

**Comment by jridgewell** [comment]
> Perhaps the bug is that the throw happens in the wrong place. Simply declaring the class should not throw; only actually attempting to use it does.

Yup.

> This might mean that even the sole working example of class X extends null { constructor () { return {} } } is also broken here.

That should work, though.

**Comment by Kovensky** [comment]
Exactly; that should work, because it basically works around the class construction algorithm.

But if the throw is happening at declaration time, it, too, will be broken by us.

**Comment by Kovensky** [comment]
```js
class X extends null {
  constructor () {
    const self = Object.create(new.target.prototype)
    self.foo = 'bar'
    return self
  }
  baz () {
    return this.foo
  }
}

const x = new X()
x.foo // 'bar'
x.baz() // 'bar'
Object.getPrototypeOf(x) // === X.prototype
Object.getPrototypeOf(Object.getPrototypeOf(x)) // === null
```

That "workaround" actually makes `extends null` "work as intended" here 😆

**Comment by jridgewell** [comment]
This was fixed by #7772.
